apps/emqx_connector/src/schema/emqx_connector_schema.erl

```diff
@@ -497,19 +514,33 @@ status_and_actions_fields() ->
             )}
     ].
 
+resource_opts_ref(Module, RefName) ->
+    [
+        {resource_opts,
+            mk(
+                ref(Module, RefName),
```

Comment:
could it be a common schema instead of each connector implement their own schema ?